test/NuGet.Core.Tests/NuGet.Packaging.Test/PackageCompatibiltyValidationTests.cs

```diff
@@ -0,0 +1,215 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using NuGet.Common;
+using NuGet.Commands;
+using NuGet.Test.Utility;
+using Xunit;
+using NuGet.Packaging.Rules;
+
+namespace NuGet.Packaging.Test
+{
+    public class PackageCompatibiltyValidationTests
+    {
+        [Fact]
+        public void NoWarningRaisedWhenBuildFound()
```

Comment:
I like the test naming convention that Damon often suggests: `<method being tested>_<scenario>_<expected result>`. For example `Validate_PackageWithBuildFilesWithoutLibOrRefFiles_ShouldNotWarn()`.